### PR TITLE
Remove py.path special handling from io.common

### LINKED ISSUE #26450: Deprecate py.path support
Since our min version is Py35 path lib is guaranteed to be installed. However we have a few functions that accept py.path.LocalPath objects for compatability

Not sure if we need an explicit deprecation cycle or if this is implied with moving minimum support to Py35

### PR DESCRIPTION
**Pull request by nandahkrishna**
- [x] closes #26450 
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by nandahkrishna** [comment]
Just an initial commit, with the DeprecationWarning added in io/common.py on py.path.local import. Please do tell mike if this is right, and what I must change/how I should proceed. Thanks.

File: pandas/io/common.py [hunk 1]
```diff
@@ -132,6 +133,8 @@ def _stringify_path(filepath_or_buffer):
     if _PATHLIB_INSTALLED and isinstance(filepath_or_buffer, pathlib.Path):
         return str(filepath_or_buffer)
     if _PY_PATH_INSTALLED and isinstance(filepath_or_buffer, LocalPath):
```
Comment (jreback) [inline pandas/io/common.py]: yeah I agree with @TomAugspurger here, I think we can blow aways all of these checks (and imports) and just
```
if hasattr(filepath_or_buffer, '__fspath__'):
    return str(filepath_or_buffer)
```
and things should *just* work

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/io/common.py (hunk 1 above)
Comment (nandahkrishna) [reply pandas/io/common.py]: I've removed the imports and checks, only the hasattr check remains.

File: pandas/io/common.py [hunk 2]
```diff
@@ -115,24 +115,8 @@ def _stringify_path(filepath_or_buffer):
     Any other object is passed through unchanged, which includes bytes,
     strings, buffers, or anything else that's not even path-like.
     """
-    try:
-        import pathlib
-        _PATHLIB_INSTALLED = True
```
Comment (jreback) [inline pandas/io/common.py]: so in order to make this pass 3.5, I think you might need to:

```
import pathlib
...

if hasattr(filepath_or_buffer, '__fspath__'):
    ....
elif isintance(filepath_or_buffer, pathlib.Path):
   return str(filepath_or_buffer)
```

Comment (nandahkrishna) [reply pandas/io/common.py]: Ah. I'll change this.

**Review by WillAyd — APPROVED** [review]
lgtm

File: pandas/io/common.py [hunk 3]
```diff
@@ -115,24 +115,11 @@ def _stringify_path(filepath_or_buffer):
...
-    except ImportError:
-        _PATHLIB_INSTALLED = False
-
-    try:
-        from py.path import local as LocalPath
-        _PY_PATH_INSTALLED = True
-    except ImportError:
-        _PY_PATH_INSTALLED = False
-
+    import pathlib
```
Comment (jreback) [inline pandas/io/common.py]: you can move this to the top of the module

**Review by jreback — CHANGES_REQUESTED** [review]
pls move the import; otherwise lgtm. ping on green.

**Comment by nandahkrishna** [comment]
The import has been moved, and all checks pass. Is any other change required?

**Comment by WillAyd** [comment]
Thanks!

**Pull request merged**